packages/xstate-react/index.v3.d.ts

```diff
@@ -1,45 +1,79 @@
 import {
-  EventObject,
-  Interpreter,
+  AnyStateMachine,
+  AreAllImplementationsAssumedToBeProvided,
+  InternalMachineOptions,
+  InterpreterFrom,
   InterpreterOptions,
-  MachineOptions,
   Observer,
-  State,
-  StateMachine,
-  Typestate
+  StateFrom
 } from 'xstate';
-import { MaybeLazy, UseMachineOptions } from './lib/types';
+import { MaybeLazy, Prop, UseMachineOptions } from './lib/types';
 
 export * from './lib/index';
 
-export declare function useMachine<
```

Comment:
I've unified signature declarations that target `TS@<4`. It was a mistake to have different generic parameters for `TS@<4` and `TS@>=4` - as upgrading TS version shouldn't immediately break somebody's code. I've fixed the issue as this release will be a major release for `@xstate/react` anyway.

Those still need to be in a separate file since labeled tuple elements were introduced in TS 4. Those declarations here work almost the same as the ones for TS 4. I didn't realize before that I could just ditch the labels and have the rest to behave the same.